### PR TITLE
feat: squad directory layout

### PR DESCRIPTION
**Pull request by ilasw**
## Changes
- Added layout for new squads directory

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://mi-488-squad-directory-navbar-la.preview.app.daily.dev/squads/discover

### CODE REVIEW

**Comment by rebelchris** [comment]
@ilasw more general, why did the lock file change :jenkins-triggered:

**Comment by ilasw** [comment]
> @ilasw more general, why did the lock file change :jenkins-triggered:

It's a typo from the try to add directory layout to the storybook, will revert shortly!

File: packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  label,
+  onClick,
+  path,
+  ...attrs
+}: SquadNavbarItemProps): ReactElement => (
+  <li
+    {...attrs}
+    className={classNames(
+      'relative py-3',
+      'after:absolute after:bottom-0 after:left-0 after:right-0 after:mx-auto after:w-14 after:border-b-2',
```
Comment (rebelchris) [inline packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx:34]: What are you adding here, like a css arrow?

File: packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        <Button
+          className="absolute right-full top-1/2 -translate-y-1/2"
+          disabled={isAtStart}
+          icon={<ArrowIcon className="-rotate-90" />}
+          onClick={onClickPrevious}
+          variant={ButtonVariant.Option}
+        />
+      )}
+      <ul
+        className="no-scrollbar laptop:3 relative flex flex-1 flex-row flex-nowrap gap-2 overflow-x-auto scroll-smooth "
```
Comment (rebelchris) [inline packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx:92]: What does `laptop:3` do?

File: packages/shared/src/lib/constants.ts [hunk 3]
```diff
@@ -67,3 +67,18 @@ export const heimdallUrl = isDevelopment
...
+  web: '/squads/discover/web',
+  mobile: '/squads/discover/mobile',
+  games: '/squads/discover/games',
+  devOps: '/squads/discover/DevOps',
+  cloud: '/squads/discover/cloud',
+  career: '/squads/discover/career',
+  data: '/squads/discover/data',
+  fun: '/squads/discover/fun',
+  devtools: '/squads/discover/devtools',
+};
```
Comment (rebelchris) [inline packages/shared/src/lib/constants.ts:84]: This is just temporary?
I assume they should be dynamic based on database categories?

File: packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    <nav
+      aria-label="Squad directory navigation"
+      className={classNames(
+        'relative -mx-4 border-b border-border-subtlest-tertiary px-4 laptop:border-0',
+        'flex-row flex-nowrap gap-2 laptop:flex',
+        className,
+      )}
+      {...attrs}
+    >
+      {!isAtStart && isOverflowing && isLaptop && (
```
Comment (rebelchris) [inline packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx:82]: More wondering should the Horizontal nav not handle the buttons already?

**Review by rebelchris — COMMENTED** [review]
More some questions for now.

File: packages/shared/src/lib/constants.ts (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/lib/constants.ts:84]: The only static ones here should be `discover` and `featured`. For the other ones, we should have a new query to fetch all categories. We can keep this temporary as I am already introducing it in a separate PR.

File: packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx (hunk 1 above)
Comment (ilasw) [reply packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx:34]: This is the border at bottom of the active element, this line has a fixed width so i made it with those classes on a `::after`

File: packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx (hunk 2 above)
Comment (ilasw) [reply packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx:92]: think is a typo, removing it!

File: packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx (hunk 4 above)
Comment (ilasw) [reply packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx:82]: The HorizontalScroll component can't be adapted to this use case because its controls are in a different row from the content (due to the layout). I’ve abstracted the controls (to be used headless) from the hook and used them there.

File: packages/shared/src/components/HorizontalScroll/useScrollManagement.tsx [hunk 5]
```diff
@@ -19,7 +19,7 @@ export const useScrollManagement = (
       const { scrollLeft, scrollWidth, clientWidth } = ref.current;
 
       setIsAtStart(scrollLeft === 0);
-      setIsAtEnd(scrollLeft + clientWidth >= scrollWidth);
+      setIsAtEnd(scrollLeft + clientWidth >= scrollWidth - 1);
```
Comment (rebelchris) [inline packages/shared/src/components/HorizontalScroll/useScrollManagement.tsx:22]: Why was this needed?

File: packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx:34]: Ah shouldn't it just be w-full of each element then?

File: packages/shared/src/lib/constants.ts (hunk 3 above)
Comment (rebelchris) [reply packages/shared/src/lib/constants.ts:84]: If you keep it just add a TODO so it doesn't get forgotten please 🙏

**Review by rebelchris — APPROVED** [review]
Nothing major blocking for me

File: packages/shared/src/components/HorizontalScroll/useScrollManagement.tsx (hunk 5 above)
Comment (ilasw) [reply packages/shared/src/components/HorizontalScroll/useScrollManagement.tsx:22]: Items inside was scrolling by width -1 and was not triggering the atEnd, really strange :\

**Comment by ilasw** [comment]
@rebelchris in figma, border is not full width, but I can do like this if we want to change it. 
image

File: packages/webapp/pages/squads/index.tsx [hunk 6]
```diff
@@ -1,46 +1,30 @@
-import React, { ReactElement, useContext } from 'react';
+import React, { ReactElement } from 'react';
```
Comment (sshanzel) [inline packages/webapp/pages/squads/index.tsx:1]: Let's revert the change here to avoid making any changes on the existing page. The deleted Card here should've been removed at the intermediate branch now.

File: packages/shared/src/components/squads/layout/useSquadDirectoryLayout.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  categoryPaths: Record<string, string>;
+  isMobileLayout: boolean;
+}
+
+export const useSquadDirectoryLayout = (): SquadDirectoryLayoutReturn => {
+  const { squads = [] } = useAuthContext();
+  const hasSquad = !!squads?.length;
+  const isLaptop = useViewSize(ViewSize.Laptop);
+  const [isMySquadsActive, setIsMySquadsActive] = useToggle(false);
+
```
Comment (sshanzel) [inline packages/shared/src/components/squads/layout/useSquadDirectoryLayout.ts:24]: There is now a `useSquadCategories` hook you can instantiate here, then map it with the `squadCategoriesPaths`.

**Review by sshanzel — CHANGES_REQUESTED** [review]
Looks great overall, don't have much to add as things looked well. Just had a few comments.

File: packages/shared/src/components/squads/layout/useSquadDirectoryLayout.ts (hunk 7 above)
Comment (ilasw) [reply packages/shared/src/components/squads/layout/useSquadDirectoryLayout.ts:24]: Okay, thanks!

File: packages/webapp/pages/squads/discover/index.tsx [hunk 8]
```diff
@@ -4,41 +4,44 @@ import { SquadHorizontalList } from '@dailydotdev/shared/src/components/cards/sq
...
-            className="h-12"
-            src={cloudinary.squads.directory.featured}
-            alt="A text containing the word 'Featured'"
-          />
-        }
-        query={{ isPublic: true, featured: true }}
-      />
-      {categories.map(({ node }) => (
+    <SquadDirectoryLayout>
+      <div className="relative flex w-full flex-col gap-6 p-5">
```
Comment (sshanzel) [inline packages/webapp/pages/squads/discover/index.tsx:15]: Nice! Non-blocking: would be possible to remove this `div` and just pass the classes directly to the `SquadDirectoryLayout`?

**Review by sshanzel — APPROVED** [review]
Nice work! That was fast! ⚡

File: packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        <Button
+          className="absolute right-full top-1/2 -translate-y-1/2"
+          disabled={isAtStart}
+          icon={<ArrowIcon className="-rotate-90" />}
+          onClick={onClickPrevious}
+          variant={ButtonVariant.Option}
```
[the remark this answers is not in the document]
Comment (ilasw) [reply packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx:89]: Yes, nice catch! Think was a typo, thanks! 🙏

**Review by nensidosari — APPROVED** [review]
Looks good, let's merge
